### DISCUSSION TITLE
Using S3 as a storage provider

### DISCUSSION THREAD

**Discussion by trajche** [vitodeploy/vito#200] [Ideas]
The idea is to enable using low cost S3 for storage provider.

Providers such as DigitalOcean or Scaleway offer this as a service, or users can use Minio to self-host.  

I believe the project already uses [flywheel's packages](https://laravel.com/docs/11.x/filesystem) for sftp and dropbox?

**Comment by saeedvaziry** [comment]
This is a bit different. Vito itself is a Laravel project which you can have S3 inside itself but StorageProviders are on your target servers actually. So we will need to implement the S3 API using cli and curl

**Reply by trajche** [reply]
Got it! Are all actions executed on target servers via ssh or is there an agent running as well?

**Reply by saeedvaziry** [reply]
They are SSH commands

**Reply by l0gical** [reply]
That would be awesome +1 for S3 support.
And thanks guys, only stumbled across this project today, am an ex Forge/Ploi user, this is pleasant to use, was real glad to see that it is Laravel based which I use extensively.
Without trying to overload, Fail2Ban would be a good feature to add.

**Reply by sinnbeck** [reply]
I would recommend using rclone as that has support for a ton of different server types including S3

https://rclone.org/s3/

https://rclone.org/docs/

This would also work with ftp and dropbox, making the StorageProviders much more similar in implementation

Oh and you can download a zip file here with a portable executable. So no need to install anything.
https://rclone.org/downloads/

**Comment by saeedvaziry** [comment]
#281
